src/runtime/link-accounts-flow-test.js

```diff
@@ -470,4 +440,117 @@ describes.realWin('LinkSaveFlow', {}, env => {
+    let messageCallback = undefined;
+    sandbox.stub(port, 'onMessage', cb => {
+      messageCallback = cb;
+    });
+    activitiesMock.expects('openIframe').returns(Promise.resolve(port));
+    return linkSaveFlow.start().then(() => {
+      messageCallback({
+        'getLinkingInfo': true,
+      });
+      return linkSaveFlow.requestPromise_;
+    }).then(() => {
+      throw new Error('must have failed');
+    }, reason => {
+      expect(() => {
+        throw reason;
+      }).to.throw(/Both authCode and token are available/);
+      return Promise.resolve();
+    });
+  });
+
+  it('should should fail if neither token nor authCode is present', () => {
+    linkSaveFlow = new LinkSaveFlow(runtime, () => {});
+    let messageCallback = undefined;
+    sandbox.stub(port, 'onMessage', cb => {
+      messageCallback = cb;
+    });
+    activitiesMock.expects('openIframe').returns(Promise.resolve(port));
+    return linkSaveFlow.start().then(() => {
+      messageCallback({
+        'getLinkingInfo': true,
+      });
+      return linkSaveFlow.requestPromise_;
+    }).then(() => {
+      throw new Error('must have failed');
+    }, reason => {
+      expect(() => {
+        throw reason;
+      }).to.throw(/Neither token or authCode is available/);
+    });
+  });
+
+  it('should respond with subscription request with token', () => {
+    const reqPromise = new Promise(resolve => {
+      resolve({token: 'test'});
+    });
+    linkSaveFlow = new LinkSaveFlow(runtime, () => reqPromise);
+    let messageCallback = undefined;
+    sandbox.stub(port, 'onMessage', cb => {
+      messageCallback = cb;
+    });
+    const messageStub = sandbox.stub(port, 'message');
+    activitiesMock.expects('openIframe').returns(Promise.resolve(port));
+    return linkSaveFlow.start().then(() => {
+      messageCallback({
+        'getLinkingInfo': true,
+      });
+      return linkSaveFlow.requestPromise_;
+    }).then(() => {
+      expect(messageStub).to.be.calledOnce.calledWith(
+        {token: 'test'});
+      return Promise.resolve();
+    });
+  });
+
+  it('should respond with subscription request with authCode', () => {
+    linkSaveFlow = new LinkSaveFlow(runtime, () => {
+      return {authCode: 'testCode'};
+    });
+    let messageCallback = undefined;
+    sandbox.stub(port, 'onMessage', cb => {
+      messageCallback = cb;
+    });
+    const messageStub = sandbox.stub(port, 'message');
+    activitiesMock.expects('openIframe').returns(Promise.resolve(port));
+    return linkSaveFlow.start().then(() => {
+      messageCallback({
+        'getLinkingInfo': true,
+      });
+      return linkSaveFlow.requestPromise_;
+    }).then(() => {
+      expect(messageStub).to.be.calledOnce.calledWith(
+        {authCode: 'testCode'});
+      return Promise.resolve();
+    });
+  });
+
+  it('should callback promise rejected should close dialog', () => {
+    linkSaveFlow = new LinkSaveFlow(runtime, () => Promise.reject('no token'));
```

Comment:
We need a similar test where the callback throws synchronously. E.g. `() => {throw new Error('failed');}`.

Reply:
done